pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -259,13 +259,13 @@ func (r *replicaRaftStorage) Term(i uint64) (uint64, error) {
 		return r.mu.lastTerm, nil
 	}
 	// Try to retrieve the term for the desired entry from the entry cache.
+	// TODO(pavelkalinnikov): maybe drop this, because term calls entries, which
```

Comment:
Sounds reasonable. The difference between `cache.Get` and `cache.Scan` isn't big; Get internally looks like Scan.

Reply:
Done even a better thing, see the commits. I figured that decoupling `term` from `entries` solves this and a few other problems.